src/reports/reportData.ts

```diff
@@ -0,0 +1,85 @@
+import * as systeminformation from "systeminformation";
+
+const UNKNOWN_SPEED = -1;
+const UNKNOWN_CORES_AMOUNT = -1;
+const UNKNOWN_MEMORY_SIZE = -1;
+
+const byteToGigabyte = (bytes: number): number => bytes / 1e9;
+
+export type OsInfo = {
+  platform: string;
+  distro: string;
+  arch: string;
+  kernel: string;
+};
+
+export type CpuInfo = {
+  manufacturer: string;
+  brand: string;
+  speedGHz: number;
+  cores: number;
+};
+
+export type Specs = {
+  os: OsInfo;
+  cpu: CpuInfo;
+  memoryGB: number;
+};
+
+export type ReportData = {
+  timestamp: string;
+  os: string;
+  kernel: string;
+  cpu: string;
+  cores: string;
+  speedGHz: string;
+  memoryGB: string;
+};
+
+let specsCache: Promise<Specs> | undefined;
+
+async function getSpecs(): Promise<Specs> {
+  const cpuData = await systeminformation.cpu();
+  const osData = await systeminformation.osInfo();
+  const memoryData = await systeminformation.mem();
+
+  return {
+    os: {
+      platform: osData?.platform ?? "unknown-platform",
+      distro: osData?.distro ?? "unknown-distro",
+      arch: osData?.arch ?? "unknown-arch",
+      kernel: osData?.kernel ?? "unknown-kernel",
+    },
+    cpu: {
+      manufacturer: cpuData?.manufacturer ?? "unknown-manufacturer",
+      brand: cpuData?.brand ?? "unknown-brand",
+      speedGHz: cpuData?.speed ?? UNKNOWN_SPEED,
+      cores: cpuData?.cores ?? UNKNOWN_CORES_AMOUNT,
+    },
+    memoryGB: memoryData?.total
+      ? Math.round(byteToGigabyte(memoryData.total))
+      : UNKNOWN_MEMORY_SIZE,
+  };
+}
+
+async function getSpecsCache(): Promise<Specs> {
+  if (!specsCache) {
+    specsCache = getSpecs();
+  }
+
+  return specsCache;
+}
+
+export default async function getReportData(): Promise<ReportData> {
+  const specs = await getSpecsCache();
+
+  return {
+    timestamp: `${new Date().getTime()}`,
```

Comment:
why there are 2 transformations( `osData` -> `Spec` -> `ReportData`)?

Reply:
`osData` is part of `Specs`, and `Specs` are not meant to be the final `ReportData` (more properties that are not spec related can be added potentially)